batch_job/src/work_queue_factory.c

```diff
@@ -381,21 +381,19 @@ static int submit_worker( struct batch_queue *queue )
 		files = newfiles;
 	}
 	
-	if(os){
+	if(runos_os){
 		char* vc3_cmd = string_format("./vc3-builder --require cctools-statics -- %s",cmd);
-		char* temp = string_format("python /afs/crc.nd.edu/group/ccl/software/runos/runos.py %s %s",os,vc3_cmd);
+		char* temp = string_format("python /afs/crc.nd.edu/group/ccl/software/runos/runos.py %s %s",runos_os,vc3_cmd);
 		free(vc3_cmd);
 		free(cmd);
 		cmd = temp;
 		temp = string_format("%s,%s",files,"vc3-builder");
 		free(files);
 		files = temp;
 	}else{
```

Comment:
Where is cmd defined for the regular case?

Reply:
cmd is only ever wrapped around if it's changed. So the root cmd is defined first by "if catalog else" statements, and then only modified from there. The else statement here for the runos_os if statement simply verifies that we're not using runos or vc3-builder, so to use the normal work_queue_worker.